lib/fuel-rest.js

```diff
@@ -97,18 +97,22 @@ FuelRest.prototype._processRequest = function(options, callback) {
 			return;
 		}
 
-		retry       = options.retry || false;
 		authOptions = clone(options.auth);
 
-		delete options.retry;
-		delete options.auth;
-
 		options.uri     = helpers.resolveUri(this.origin, options.uri);
 		options.headers = merge({}, this.defaultHeaders, options.headers);
-		options.headers.Authorization = options.headers.Authorization || 'Bearer ' + authResponse.accessToken;
+
+		if(!options.headers.Authorization) {
+			options.headers.Authorization = 'Bearer ' + authResponse.accessToken;
+			retry = options.retry || false;
```

Comment:
At first I wondered, but now I think it's fine. Since the auth header shouldn't really diff from the one provided (just there for an extra option), this is totally cool. :+1: